coil-base/src/main/java/coil/request/Metadata.kt

```diff
@@ -0,0 +1,19 @@
+@file:OptIn(ExperimentalCoilApi::class)
+
+package coil.request
+
+import coil.annotation.ExperimentalCoilApi
+import coil.decode.DataSource
+import coil.memory.MemoryCache
+
+/**
+ * Supplemental information about a successful image request.
+ *
+ * @param key The cache key for the image in the memory cache.
+ *  It is null if the image was not written to the memory cache.
+ * @param dataSource The data source that the image was loaded from.
+ */
+data class Metadata(
```

Comment:
Not sure `Metadata` is the best name for this... Created this type to be able to return more things as needed in the future through `ImageRequest.Listener.onSuccess`.

Reply:
Makes sense to me. I could see there being a nice feature too where you can pass raw data via the request and it gets returned in the metadata. Something like 
```
val request = LoadRequest.Builder(context)
    .data("https://www.example.com/image.jpg")
    .additionalData(someObject)
```
similar to what you can do with Android views via `setTag`, and this gets returned via `Metadata` (without type safety). Just a potential idea, not a change request to this PR 😄

Reply:
Good point! OkHttp has a `tag` system as well. Might be a good addition...